python/lsst/pipe/tasks/visualizeVisit.py

```diff
@@ -0,0 +1,250 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import numpy as np
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.afw.math as afwMath
+import lsst.afw.image as afwImage
+import lsst.afw.cameraGeom.utils as afwUtils
+
+
+__all__ = ['VisBinConfig', 'VisBinTask', 'VisMosaicConfig', 'VisMosaicTask']
+
+
+class VisBinConnections(pipeBase.PipelineTaskConnections,
+                        dimensions=("instrument", "detector")):
+    inputExp = cT.Input(
+        name="calexp",
+        doc="Input exposure data to mosaic.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector"),
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        doc="Input camera to use for mosaic geometry.",
+        storageClass="Camera",
+        dimensions=("instrument", "calibration_label"),
+    )
+
+    outputExp = cT.Output(
+        name="calexpBin",
+        doc="Output binned image.",
+        storageClass="ExposureF",
+        dimensions=("instrument", "detector"),
+    )
+
+
+class VisBinConfig(pipeBase.PipelineTaskConfig,
+                   pipelineConnections=VisBinConnections):
+    """Configuration for focal plane visualization.
+    """
+    binning = pexConfig.Field(
+        dtype=int,
+        default=8,
+        doc="Binning factor to apply.",
+    )
+    detectorKeyword = pexConfig.Field(
+        dtype=str,
+        default='DET-ID',
+        doc="Metadata keyword to use to find detector if not available from input.",
+    )
+
+
+class VisBinTask(pipeBase.PipelineTask,
+                 pipeBase.CmdLineTask):
+    """Task for focal plane visualization.
+    """
+    ConfigClass = VisBinConfig
+    _DefaultName = 'VisBin'
+
+    def run(self, inputExp, camera):
+        """Bin input image, attach associated detector.
+
+        Parameters
+        ----------
+        inputExp : `lsst.afw.image.Exposure`
+            Input exposure data to bin.
+        camera : `lsst.afw.cameraGeom.Camera`
+            Input camera to use for mosaic geometry.
+
+        Returns
+        -------
+        output : `lsst.pipe.base.Struct`
+            Results struct with attribute:
+
+            ``outputExp``
+                Binned version of input image. (`lsst.afw.image.Exposure`)
+        """
+        if inputExp.getDetector() is None:
+            detectorId = inputExp.getMetadata().get(self.config.detectorKeyword)
+            if detectorId is not None:
+                inputExp.setDetector(camera[detectorId])
+
+        binned = inputExp.getMaskedImage()
+        binned = afwMath.binImage(binned, self.config.binning)
+        outputExp = afwImage.makeExposure(binned)
```

Comment:
if this is going to be an exposure, you should copy all the associated info from an exposure, ie the exposure info and such.